src/platforms/esp32/components/avm_builtins/uart_driver.c

```diff
@@ -443,13 +509,12 @@ static NativeHandlerResult uart_driver_consume_mailbox(Context *ctx)
         if (is_closed) {
             if (UNLIKELY(memory_ensure_free(ctx, TUPLE_SIZE(2) * 2 + REF_SIZE) != MEMORY_GC_OK)) {
```

Comment:
This could be changed to `memory_ensure_free_with_roots`, and include ref as a root, reduce allocation by REF_SIZE and avoid to convert ref back and forth to uint64_t.